src/handlers/scheduledEventHandler.ts

```diff
@@ -141,3 +147,52 @@ export const syncIdle7dUsers = async (env: env) => {
 export const syncOnboarding31dPlusUsers = async (env: env) => {
 	return await apiCaller(env, 'discord-actions/group-onboarding-31d-plus', 'PUT');
 };
+
+export async function filterOrphanTasks(env: env) {
+	const namespace = env[NAMESPACE_NAME] as unknown as KVNamespace;
+	let lastOrphanTasksFilteration: string | null = '0';
+	try {
+		lastOrphanTasksFilteration = await namespace.get('ORPHAN_TASKS_UPDATED_TIME');
+		if (lastOrphanTasksFilteration === null) {
+			console.error('Error while fetching KV "ORPHAN_TASKS_UPDATED_TIME" timestamp');
+		}
+		if (!lastOrphanTasksFilteration) {
+			lastOrphanTasksFilteration = '0';
+		}
```

Comment:
- as we're using the `!` operator console.error can be moved to the if statement below.
- if we're setting the value of `lastOrphanTasksFilteration` in the last if statement why are we logging an error above?
- for instances where the value is set for the first time for the namespace the value returned from `namespace.get` will be null in which case creating an error log does not seem to be the right thing to do, what do you think?

Reply:
1st point: Makes sense making the change.
2nd Point: if there is some issue while getting the `lastOrphanTasksFilteration` but makes sense to put it in second if statement
3rd point: Yes right but what in the cases when we have the timestamp saved but something went wrong?

Reply:
Made the changes.

Reply:
>2nd Point: if there is some issue while getting the lastOrphanTasksFilteration but makes sense to put it in second if statement
3rd point: Yes right but what in the cases when we have the timestamp saved but something went wrong?

we already have a catch block for it if something goes wrong while fetching data the catch block will be triggered.